From b3a30b4faaf6952ee8407d871892a86211cef04f Mon Sep 17 00:00:00 2001
From: Rinigus <rinigus.git@gmail.com>
Date: Sat, 15 Jun 2024 12:07:31 +0300
Subject: [PATCH] Update protobuf configuration

---
 CMakeLists.txt | 21 ++++++++-------------
 1 file changed, 8 insertions(+), 13 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index fa47a39dc..b632423eb 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -179,22 +179,17 @@ else()
   message(STATUS "Using curl from the outside")
 endif()
 
-if(NOT Protobuf_FOUND)
-  find_package(Protobuf REQUIRED)
-endif()
-
-message(STATUS "Using pbf headers from ${PROTOBUF_INCLUDE_DIR}")
-message(STATUS "Using pbf libs from ${PROTOBUF_LIBRARY}")
-message(STATUS "Using pbf release libs from ${PROTOBUF_LIBRARY_RELEASE}")
-message(STATUS "Using pbf debug libs from ${PROTOBUF_LIBRARY_DEBUG}")
+# Protobuf is non-trivial to include via pkg-config, pkg_check_modules has no way to check
+# for protoc location in a platform agnostic manner
+# newer protobuf versions require a compat bool
+set(protobuf_MODULE_COMPATIBLE ON CACHE BOOL "")
+find_package(Protobuf REQUIRED CONFIG)
+message(STATUS "Using protoc from ${Protobuf_PROTOC_EXECUTABLE}")
+message(STATUS "Using pbf headers from ${Protobuf_INCLUDE_DIRS}")
+message(STATUS "Using pbf libs from ${PROTOBUF_LIBRARIES}")
 if(TARGET protobuf::libprotobuf-lite)
   message(STATUS "Using pbf-lite")
 endif()
-
-# Allow linking against full or lite version of Protobuf library
-# TODO: After switching to CMake 3.12+, replace with
-#       $<TARGET_EXISTS:protobuf::libprotobuf
-#       $<TARGET_EXISTS:protobuf::libprotobuf-lite
 if(TARGET protobuf::libprotobuf-lite)
   set(valhalla_protobuf_targets protobuf::libprotobuf-lite)
 elseif(TARGET protobuf::libprotobuf)
-- 
2.44.2

